### DISCUSSION TITLE
Antmedia Adaptive Bitrate Quality Problem

### DISCUSSION THREAD

**Discussion by onur-ardal** [ant-media/Ant-Media-Server#6631] [Q&A]
Hi everyone,
I am curious about adaptive bitrate work mechanism. First of all, if I am misunderstood, could you explain it?

Webrct has own encoding mechanism on client side(This client it could be android, iOS or chrome). When I send video stream to ams with h264 encoding, it is already be compressed. 
Then I open Adaptive bitrate, ffmpeg compress again my already compressed video. So my observation is a lot of quality loss.

Otherwise if the client sends it without encoding it (I haven't seen an example of this in clients) or bigger resolution and bitrate, won't it have problems broadcasting when there is a client with low internet speed?

I want to discuss it.

**Comment by elijahwere** [comment]
Ant Media server breaks down a video into multiple resolutions and then uses its adaptive bitrate streaming algorithm to serve the right resolution or quality based on the end-user's network performance. 


[How can I enable Adaptive Bitrate streaming in Ant Media Server?](https://www.ugacomp.com/how-can-i-enable-adaptive-bitrate-streaming-in-ant-media-server/)

**Reply by onur-ardal** [reply]
Yes, I know how works adaptive bitrate. But When antmedia create multiple resolutions, that resource video already compressed from Webrtc h264. So this reason in my opinion live stream loss quality.

**Reply by yashtandon113** [reply]
Hi @onur-ardal 
I hope you are doing well. Let me try to explain.

For example, if you are sending the H264 encoded stream via browser with 720p resolution, and if you want to deliver it with multiple resolutions, then yes, you have to use the adaptive bitrate, which will further compress the video into different bitrates and resolutions.

According to the resolution, the quality will be different. If you are sending with 720p, then it is recommended to use ABRs 720p or below for a better transcoding experience.

If the resolution and bitrate of your stream are not constant and varying, then you may still face the quality loss.

Could you please share the step-by-step reproduce scenario to explain the quality loss using ABRs? We will try to replicate the same and investigate further.

**Comment by onur-ardal** [comment]
Hi @yashtandon113,

Thank you for concern. 
Our project is 1 to n live-streaming for live auction. We are using flutter for mobile client. 

**First scenario;**
- Mobile client has 1280*720 resolution and 30fps.
- Antmedia disabled ADR.

Video quality is acceptable. 
But we have older Webrtc version in flutter client, and it doesn't have "set bitrate configuration". But we want to under control bitrate.
So we decided to use ADR.

**Second scenario;**
- Mobile client has 1280*720 resolution and 30fps.
- Antmedia activated ADR only one resolution (720p - 2000 bitrate)

We noticed video quality loss. Then i was think, it could be architectural problem. Because no matter what resolution I send if ADR is open, ffmpeg compress again.

Maybe If I set below scenario, I get better video quality. But this time if publisher has not good internet, there maybe interruptions in broadcasting.

**Third scenario;**
- Mobile client sent 1920*1080 resolution and 30fps
- Antmedia ADR 720p - 2000 bitrate

**Fourth scenario;**
- Mobile client sent 1280*720 resolution
- Antmedia ADR 
          - 720p - 2000 bitrate (loss quality)
          - 540p - 1000 bitrate (accetable quality. Because source video is bigger)

Maybe ADR should not recompress the video resolution that I sent the largest. It should compress the other resolutions anyway.

**Reply by yashtandon113** [reply]
@onur-ardal 

I tested the scenario in which I published a 640x480 stream from the Flutter SDK in landscape mode while enabling 480 ABR on the server side. I played the stream using WebRTC and saw no significant quality loss, and in my observation the quality was fine at 480p with 1000 Kbps.

One thing to check is that if you are sending the stream at 30 FPS, ensure that the 30 FPS is also enabled for ABR on the server side and not more than that.

**Maybe ABR should not compress the video resolution that I sent the largest. It should compress the other resolutions anyway.**

Let me discuss with my colleague whether this is possible through any property or not.
